### ISSUE TITLE
Incompatibility issues, Gun lag, heads and other things do not turn...

### ISSUE THREAD

**Issue by InputSoldatcc** [studio-minus/madness-interactive-reloaded#38]
**Describe the bug**
The bugs are plentiful;
1. Everyone is an amputee
2. Heads do not turn
3. Guns fall through the floor (I doubt they even have collision)
4. decapitation
5. Lag when someone has a gun

**Steps to reproduce**
1. Launch game
2. No game

**Expected behavior**
Game

**Screenshots or video**
![image](https://github.com/user-attachments/assets/ff37f73b-8db7-40c3-9f4f-923dd2d7b1f9)
The inexplicable "Main Menu™"

![image](https://github.com/user-attachments/assets/a06550d3-33c2-4a69-a282-15d474f640b9)
Main menu errors

![image](https://github.com/user-attachments/assets/4498bc47-3d1f-48fc-b856-2b7a06cca630)
Thy Wardrobe

![image](https://github.com/user-attachments/assets/6c9b77de-9fc8-4434-8680-a4def240cbe7)
The terrifying image of death. (Oh look he has arms now!), also crashes the game when reviving

![image](https://github.com/user-attachments/assets/88eca004-20b1-484a-91c0-fd5bff2bf0b1)
The crash output in question after trying to revive

![image](https://github.com/user-attachments/assets/02eb0cc6-22c9-40e8-bc83-4778a35265be)
How come he has arms but I dont?

![image](https://github.com/user-attachments/assets/93cb5c58-fdbd-4550-ac1e-ab849e160bd5)
Upon entering a level where someone has a gun, the fps reaches an all time low

**Desktop (please complete the following information):**
Windows 11
Build 22631.4037
MIR Versions: Up to 0.49.1

**Additional context**
None

**Comment by mestiez** [comment]
Wow that's spectacular. What kind of computer do you have?

**Comment by InputSoldatcc** [comment]
Wouldnt qualify as a good PC thats for sure, I have another GPU, Ill test the game with it tomorrow morning if I remember to.

Oh also, two grunts spawned with a single hand.
![image](https://github.com/user-attachments/assets/8945bd8e-1a9e-4044-9830-0fbc01846a72)

**Comment by InputSoldatcc** [comment]
tried going into a level with grunts spawning with guns by default

![image](https://github.com/user-attachments/assets/93cb5c58-fdbd-4550-ac1e-ab849e160bd5)
oof

**Comment by leycarnet** [comment]
> **Describe the bug** The bugs are plentiful;
> 
> 1. Missing textures, I assume this is the cause of the other bugs
> 2. The levels have succumbed to the void
> 3. Everyone is an amputee
> 4. Guns fall through the floor (I doubt they even have collision)
> 5. decapitation
> 6. Lag when someone has a gun
> 
> **Steps to reproduce**
> 
> 1. Launch game
> 2. No game
> 
> **Expected behavior** Game
> 
> **Screenshots or video** ![image](https://private-user-images.githubusercontent.com/140076004/365293191-ff37f73b-8db7-40c3-9f4f-923dd2d7b1f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTgwNjIsIm5iZiI6MTcyNTcxNzc2MiwicGF0aCI6Ii8xNDAwNzYwMDQvMzY1MjkzMTkxLWZmMzdmNzNiLThkYjctNDBjMy05ZjRmLTkyM2RkMmQ3YjFmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNDAyNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMGNmMTlhOGM0MTA3MGFkOWNjYzZiNDAxYjM5OGQ1NDc1MTEzMGZhM2Q3N2FlYWFhMmNhN2UzOTBiZGY1Y2ZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.U2dXxODnUd27OcoEVB2S_T5egfP2HEvMQjBEZ1Gn5Lo) The inexplicable "Main Menu™"
> 
> ![image](https://private-user-images.githubusercontent.com/140076004/365293123-a06550d3-33c2-4a69-a282-15d474f640b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTgwNjIsIm5iZiI6MTcyNTcxNzc2MiwicGF0aCI6Ii8xNDAwNzYwMDQvMzY1MjkzMTIzLWEwNjU1MGQzLTMzYzItNGE2OS1hMjgyLTE1ZDQ3NGY2NDBiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNDAyNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NjhmOGZiNWVlOTU4ZTAwYzQ1MzAyNWE5Nzc4OTg0MGRhYzViNmIxODU1MTdhNjlmNGI0MWM3OTBjMDQ4NTczJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jOYMYhy-q-l3Pe40vCpGY4yddA08nHvfQbqBvSBT92M) Main menu errors
> 
> ![image](https://private-user-images.githubusercontent.com/140076004/365293445-4498bc47-3d1f-48fc-b856-2b7a06cca630.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTgwNjIsIm5iZiI6MTcyNTcxNzc2MiwicGF0aCI6Ii8xNDAwNzYwMDQvMzY1MjkzNDQ1LTQ0OThiYzQ3LTNkMWYtNDhmYy1iODU2LTJiN2EwNmNjYTYzMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNDAyNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNDA1MjMzMzQwYjI3YmExZGQ4ZWE2ZDhlNmU4YzZjY2ZiNjNiNjBkOWNhYjI5ZjMxNTIxZGE4NDIwYjQ1MTNlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.S9Ui2EpkD52QXYOVSsaHFrRjg7yEsWl0_ngjQs8O_1A) Thy Wardrobe
> 
> ![image](https://private-user-images.githubusercontent.com/140076004/365293756-706aaf85-5b72-4e3f-b571-b379d7bd8949.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTgwNjIsIm5iZiI6MTcyNTcxNzc2MiwicGF0aCI6Ii8xNDAwNzYwMDQvMzY1MjkzNzU2LTcwNmFhZjg1LTViNzItNGUzZi1iNTcxLWIzNzlkN2JkODk0OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNDAyNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNTNjM2Y3YmJmYjI3MGE4ZDI1MGI2YmEzMzZmZGM5YmQyYjJlNWE5M2M1ZDk5ZTlkOWQ3NzFjMDAyMDIzNDE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.blD-gXlfRJfD3T-FK33_e8W52ACdb-GlHlmfyh-PMD4) The Void of Experiment Mode. (AKA the Nowhere in madness terms)
> 
> ![image](https://private-user-images.githubusercontent.com/140076004/365292722-6c9b77de-9fc8-4434-8680-a4def240cbe7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTgwNjIsIm5iZiI6MTcyNTcxNzc2MiwicGF0aCI6Ii8xNDAwNzYwMDQvMzY1MjkyNzIyLTZjOWI3N2RlLTlmYzgtNDQzNC04NjgwLWE0ZGVmMjQwY2JlNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNDAyNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YTk3NzNkOWI1ODFmMjJkMmIwZTBiOWNiN2UwZjE2NGVmNTE3ZDVmODViNDBjZTYyNzBmODhhYjY2YmRjMTdmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LqTppd4TUwkoGDOMo1HFHhalbaDW1TMrMUEdOmS8d24) The terrifying image of death. (Oh look he has arms now!)
> 
> ![image](https://private-user-images.githubusercontent.com/140076004/365294503-02eb0cc6-22c9-40e8-bc83-4778a35265be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTgwNjIsIm5iZiI6MTcyNTcxNzc2MiwicGF0aCI6Ii8xNDAwNzYwMDQvMzY1Mjk0NTAzLTAyZWIwY2M2LTIyYzktNDBlOC1iYzgzLTQ3NzhhMzUyNjViZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNDAyNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YzZjOWU5YTBiNjJiZWNjN2Q4YzhhYWU0YjQwZjMzNWZkNzIxMjMwZGMyZmFkNzI4YWI4N2YyOWZkYzU1YjIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EjWEZPGWl1qj1VNiAzs09I3prIRpyqqF16sZiDadmIs) How come he has arms but I dont?
> 
> ![image](https://private-user-images.githubusercontent.com/140076004/365298346-93cb5c58-fdbd-4550-ac1e-ab849e160bd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTgwNjIsIm5iZiI6MTcyNTcxNzc2MiwicGF0aCI6Ii8xNDAwNzYwMDQvMzY1Mjk4MzQ2LTkzY2I1YzU4LWZkYmQtNDU1MC1hYzFlLWFiODQ5ZTE2MGJkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNDAyNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYjAyMTViYWYxOGE5NGEwODk5YzQ2NTRkNWVhOGQxNDJkZjhhNGQ0NDZjMWIwMTU1ZTFkMWY1ZWIzNTkwNjM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2nWoZa6qO0dKcym2bXUO1PyeP0Ut-cAaF9gjl3DrJLs) Upon entering a level where someone has a gun, the fps reaches an all time low
> 
> **Desktop (please complete the following information):** Windows 11 Build 22631.4037
> 
> **Additional context** None

same for me 
update didnt fixed it

**Comment by InputSoldatcc** [comment]
Atleast theres someone else with this bug. The other gpu didn't fix it too, nor did updating .NET, it might have something to do with the compression mode?

**Comment by InputSoldatcc** [comment]
as of 0.48.5 the voided levels have been fixed

**Comment by InputSoldatcc** [comment]
![image](https://github.com/user-attachments/assets/692467dd-f5b1-42ea-b17f-55b96e1f3fc8)
If I place my mouse near my character while holding a gun, I can see my hands, but only can point up and down.

![image](https://github.com/user-attachments/assets/377a8901-f93a-4706-8e41-2de3f8889365)
The reason some grunts spawn with a single hand is because they are carrying a melee weapon

**Comment by InputSoldatcc** [comment]
updates are not fixing it 😢

**Comment by SwissCheese047** [comment]
my game lags out when i pick up a gun or melee weapon, and when i spawn in a weapon and it drops on the ground then it disappears and i cant pick it up

https://github.com/user-attachments/assets/8de2ad4d-b4d4-41c7-a5d6-3257fd6277c5

**Comment by InputSoldatcc** [comment]
ive noticed that no one added the crash output after dying so ive added it to the first message

**Comment by mestiez** [comment]
Oh! This is definitely because the hands are positioned as NaN which propagates to fuck knows where and may cause crashing. This also causes the raycast to throw because it's in NaN space. And it goes without saying that they don't render if they are in an invalid position.

What causes NaN? Division by zero, most likely. So look for that.

**Comment by InputSoldatcc** [comment]
> Oh! This is definitely because the hands are positioned as NaN which propagates to fuck knows where

https://madnesscombat.fandom.com/wiki/The_Nowhere

**Comment by InputSoldatcc** [comment]
Right so I also forgot to mention that the head doesnt turn at all, only when an animation plays. that may mean something.

**Comment by InputSoldatcc** [comment]
> Oh! This is definitely because the hands are positioned as NaN which propagates to fuck knows where and may cause crashing. This also causes the raycast to throw because it's in NaN space. And it goes without saying that they don't render if they are in an invalid position.
> 
> What causes NaN? Division by zero, most likely. So look for that.

I have a question, is the same NaN value that is causing the hands to enter the void also affecting the head's (and the Saviour shield's) ability to turn? Because I don't see any other reason why they wouldnt rotate, unless they are tied with the hand's rotation itself.

**Comment by InputSoldatcc** [comment]
OK now im really confused
![image](https://github.com/user-attachments/assets/da9f0826-08ba-4bc1-9b9b-23c2235e8e00)

Is poseParams.Character.AimDirection supposed to be like that?

**Comment by mestiez** [comment]
NaN tends to propagate all over the place. It's likely that the same division by zero is causing the head to be positioned incorrectly.

**Comment by mestiez** [comment]
I am mainly confused by the inconsistency of this NaN. Why does it only happen on some systems?

**Comment by InputSoldatcc** [comment]
> NaN tends to propagate all over the place. It's likely that the same division by zero is causing the head to be positioned incorrectly.

I found out that the `FistFight` method isn't the one causing NaN, its actually `UnarmedProtective` from what I'm seeing. Both `pos1` and `pos2` are fine same for the Hands `PosePosition` vector2s, untill the very end of the `FistFight` method where `UnarmedProtective` is called. It causes the Y value of both hand pose positions to become NaN.

**Comment by InputSoldatcc** [comment]
Update: After removing (commenting out) the part where `FistFight` calls `UnarmedProtective`, the hands are now visible and correctly positioned. However, the direction issue persists, and I still dont know what is causing the Y value of the hand vector2 to get messed up ):

**Comment by InputSoldatcc** [comment]
> I am mainly confused by the inconsistency of this NaN. Why does it only happen on some systems?

Many I've talked with about this bug have mentioned that they have a Pentium CPU or some type of Intel CPU.. Knowing intel it might be the issue.

**Comment by InputSoldatcc** [comment]
`Utilities.Lerp` causes the Y value of the Hand pose position vector2 to become NAN.

**Comment by leycarnet** [comment]
> > I am mainly confused by the inconsistency of this NaN. Why does it only happen on some systems?
> 
> Many I've talked with about this bug have mentioned that they have a Pentium CPU or some type of Intel CPU.. Knowing intel it might be the issue.

I have amd 
Amd athlon ii x4 645, its also an old one

**Comment by InputSoldatcc** [comment]
> > > I am mainly confused by the inconsistency of this NaN. Why does it only happen on some systems?
> > 
> > 
> > Many I've talked with about this bug have mentioned that they have a Pentium CPU or some type of Intel CPU.. Knowing intel it might be the issue.
> 
> I have amd Amd athlon ii x4 645, its also an old one

Thanks for your input man :D

**Comment by InputSoldatcc** [comment]
Alright so; It all ties up in the end so keep reading

`CharacterComponent.AimDirection.Y` is set to `PositiveInfinity`.

`yOffset` var in `UnarmedProtective` is set to the Y value of `AimDirection`, and it plays a major role in calculating the Y value in the `targetPos`. Since `AimDirection.Y` is infinite, you probably know where this is going. `targetPos`.Y is going to be infinite, common sense.

I believe that the `Utilities.Lerp` function does not know how to (obviously...) interpolate a normal Y value (`position`'s Y value) to an infinite Y value (`targetPos`'s value) , which in the results in a NaN Y value for the final interpolated `position` Vector2.

And even after taking out the `Utilities.Lerp` function from `UnarmedProtective`, it still going to launch the position of the hands into oblivion, since the Y value is infinite, unless you do not redefine `position`.

In summary;
The true mastermind property that causes `Utilities.Lerp`, and many others to leave scratching their heads is the one and only `CharacterComponent.AimDirection.Y`. I will investigate WHY `AimDirection.Y` is infinite and I will report back. Thank you for reading.

**Comment by mestiez** [comment]
`CharacterComponent.AimDirection` cannot be set because it is a getter property. It's set to 
```cs
Vector2.Normalize(
        AimTargetPosition - 
        new Vector2(Positioning.GlobalCenter.X, Positioning.Head.GlobalPosition.Y)
    );
```

and now everything makes sense because trying to normalise a vector with no length is a bad idea. Is it possible that `AimTargetPosition` is equal to `new Vector2(Positioning.GlobalCenter.X, Positioning.Head.GlobalPosition.Y)` at some point? I don't remember, but it seems to be so.

**Comment by InputSoldatcc** [comment]
Ive found this while searching
https://github.com/dotnet/runtime/issues/99391

It seems that the issue is actually CPU related, no wonder its persistent on some systems /:

**Comment by InputSoldatcc** [comment]
> Ive found this while searching [dotnet/runtime#99391](https://github.com/dotnet/runtime/issues/99391)
> 
> It seems that the issue is actually CPU related, no wonder its persistent on some systems /:

This only applies to Vector2.Normalize(). I have manually normalized the `AimDirection` Vector2 and now melee combat is working. However I am not done, theres still some stuff left to be fixed.

**Comment by InputSoldatcc** [comment]
Everything related to the hands has been fixed. The only issues that remain are:

No weapon collision 
No bullet tracers
Ragdolls snap into the middle of the map
Lag when someone has a gun

As for the hands, I don't know what to do, create a fork with handfix? maybe I should, I may get some feedback off of it and see if the issues i listed are persistent and not dependent on the CPU.

The choice is for you to make zooi

**Comment by InputSoldatcc** [comment]
> Everything related to the hands (well in MY build, which I have been working on since 0.48.7, i have to update my local repo) has been fixed. The only issues that remain are:
> 
> No weapon collision No bullet tracers Ragdolls snap into the middle of the map Lag when someone has a gun
> 
> As for the hands, I don't know what to do, create a fork with handfix? maybe I should, I may get some feedback off of it and see if the issues i listed are persistent and not dependent on the CPU.
> 
> The choice is for you to make zooi

On another note: I don't think manually normalizing the Vector2s instead of the built-in method is going to cause much performance issues (on older systems), because only CPUs that support SSE4.2 are the only ones that are going to benefit from the performance improvement with `Vector2.Normalize`.

**Comment by InputSoldatcc** [comment]
I have made a public fork for everyone to try, it fixes `AimDirection` and other properties using `Vector2.Normalize`.

However there is still a problem. VerletPhysics system. It tries to Normalize a NaN Vector2 (m vector2). I don't know why this happens, and if it has to do with the CPU.

**Comment by InputSoldatcc** [comment]
It just came to my mind that walgelijk probably probably has some built in methods using Vector2.Normalize, if you could tell me what they are I would be grateful :D

**Comment by mestiez** [comment]
this is a bit worrying

**Comment by mestiez** [comment]
it goes without saying that the game and game engine use `Vector2(and friends).Normalize` _all_ over the place. it being broken on obscure CPUs is kind of scary. 

i dont think using a local implementation of Normalize is a great solution. it seems like the dotnet team have already solved the issue and the fix is now in staging. i think its better to wait for this update, or even update to .NET 9 (i know its not ready, but it might be worth it).

**Comment by InputSoldatcc** [comment]
Lets just hope .NET 9 or the next .NET 8 update fixes this.. Untill then I don't think there is any other actual solution.

**Comment by leycarnet** [comment]
> Lets just hope .NET 9 or the next .NET 8 update fixes this.. Untill then I don't think there is any other actual solution.

should i update it myself or?? the problem is still there on the last version

**Comment by InputSoldatcc** [comment]
as i said i DO have a public fork which does fix the problem at (depends on your cpu) performance losses

**Comment by InputSoldatcc** [comment]
The newest .NET 9 preview SDK fixes all issues mentioned here, I guess we can wait untill its fully out then, so this issue is just going to fix itself pretty soon.

**Comment by mestiez** [comment]
.NET 9 is published and the game & engine have been updated!

**Comment by InputSoldatcc** [comment]
Awesome!
